src/duckxiterator.hpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * Under MIT license
+ * Author: Amir Mohamadi (@amiremohamadi)
+ * DuckX is a free library to work wirh docx files.
+ */
+
+#ifndef DUCKXITERATOR_H
+#define DUCKXITERATOR_H
+
+namespace duckx
+{
+	template <class T>
+	class Iterator {
+	private:
+		using ParentType = decltype(T::parent);
+		using CurrentType = decltype(T::current);
+        ParentType parent{0};
+        CurrentType current{0};
```

Comment:
Spaces vs tabs

Reply:
Unfortunately I am manually formating the code to be in the duckx style. Please provide a clang-format file if you wish, then I can automatically format it.